build-info-client/src/main/java/org/jfrog/build/client/ArtifactoryHttpClient.java

```diff
@@ -278,4 +285,10 @@ private String getResponseEntityContent(HttpEntity responseEntity) throws IOExce
         }
         return "";
     }
+
+    public void addAccessTokenHeaderToRequestIfNeeded(AbstractHttpMessage httpRequest) {
```

Comment:
This method seems to be invoked from many places in the code. This is different though when setting the username and password. In other words, when using a username/password, we don't need to worry about setting them from each and every request. This approach is easier to maintain and reduces the risk for a bug.
Have a look at the *PreemptiveHttpClient* class, which includes the core http client functionality.